packages/core/src/components/Chip/index.tsx

```diff
@@ -77,34 +81,36 @@ export class Chip extends React.Component<Props & WithStylesProps> {
         )}
         {...props}
       >
-        {(before || profileImageSrc) && (
+        {(beforeIcon || profileImageSrc) && (
           <div {...css(styles.chipItem, styles.sideContent)}>
             <div {...css(styles.sideContentInner)}>
               {profileImageSrc && <ProfilePhoto imageSrc={profileImageSrc} title="" size={4} />}
-              {before && before}
+              {beforeIcon && beforeIcon}
             </div>
           </div>
         )}
 
         <div {...css(styles.chipItem, styles.content)}>{children}</div>
 
-        {icon && (
-          <div {...css(styles.chipItem, styles.sideContent)}>
-            <div {...css(styles.sideContentInner, styles.iconWrapper)}>
-              {onIconClick ? (
-                <ButtonOrLink
-                  {...css(styles.iconButton, disabled && styles.iconButton_disabled)}
-                  disabled={disabled}
-                  onClick={onIconClick}
-                >
-                  {icon}
-                </ButtonOrLink>
-              ) : (
-                icon
-              )}
+        {afterIcon ||
+          (icon && (
```

Comment:
Condition here looks wrong.

Reply:
Maybe just set it above as a variable? `const afterIcon = props.afterIcon || props.icon`.